service/src/test/java/bio/terra/cbas/runsets/monitoring/TestSmartRunsPoller.java

```diff
@@ -198,4 +199,58 @@ void updateNewlyCompletedRuns() throws Exception {
         COMPLETE,
         actual.stream().filter(r -> r.id().equals(completedRunId)).toList().get(0).status());
   }
+
+  @Test
+  void updatingOutputFails() throws Exception {
```

Comment:
TOL/Nit 2: Is it possible to restructure the flow of the updater method so that we can test various cases of the "record updates can succeed or fail independently of each other" behavior without having to spin up all this infrastructure of an API test each time?

Reply:
I like the idea of updating this logic so we can test various scenarios easily. But I think refactoring this code would also require refactoring `updateRuns` which might be non-trivial amount of work and can be done in another PR/ticket.

Reply:
I agree!